File: Client/Assets/TrisAssets/Scripts/Commun/Objects/Identifiable.cs
namespace TrisAssets.Scripts.Logic {
    public abstract class Identifiable {
        private int ID { get;  }
        protected Identifiable(int id) {
            ID = id;
        }

        private static int _idClientSide = -10000;

        protected Identifiable() {
            ID = _idClientSide++;
        }
    }
}

File: Server/RisicardsCommun/src/issou/logic/objects/utils/Identifiable.java
package issou.logic.objects.utils;

import com.smartfoxserver.v2.entities.data.SFSObject;
import com.smartfoxserver.v2.protocol.serialization.SerializableSFSType;

public abstract class Identifiable implements SerializableSFSType {

    private static int actualId = 1;
    private final int id;

    public Identifiable(){
        this.id = actualId++;
    }

    public SFSObject toSFSObject(){
        SFSObject obj = new SFSObject();
        obj.putInt("id", id);
        return obj;
    }
    public int getId() {
        return this.id;
    }
}
